src/_igraph/graphobject.c

```diff
@@ -5130,6 +5130,71 @@ PyObject *igraphmodule_Graph_get_all_shortest_paths(igraphmodule_GraphObject *
   igraph_vector_int_list_destroy(&res);
   return list ? list : NULL;
 }
+/** \ingroup python_interface_graph
+ * \brief Calculates the k-shortest paths from/to a given node in the graph
+ * \return a list containing the k-shortest paths from/to the given node
+ * \sa TODO I don't know what to write here : igraph_get_shortest_paths
+ */
+PyObject *igraphmodule_Graph_get_k_shortest_paths(igraphmodule_GraphObject *
+                                                    self, PyObject * args,
+                                                    PyObject * kwds)
+{
+  static char *kwlist[] = {"k", "from", "to", "weights", "mode", NULL };
+  igraph_vector_int_list_t res;
+  igraph_vector_t *weights = 0;
+  igraph_neimode_t mode = IGRAPH_OUT;
+  igraph_integer_t from;
+  igraph_integer_t k;
+  igraph_integer_t to;
+  PyObject *list, *from_o, *mode_o=Py_None, *to_o=Py_None, *weights_o=Py_None,*k_o=Py_None;
+
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|OOO", kwlist, &k_o, &from_o,
+        &to_o, &weights_o, &mode_o))
+    return NULL;
+
+  if (igraphmodule_PyObject_to_neimode_t(mode_o, &mode))
+    return NULL;
+
+  if (igraphmodule_PyObject_to_integer_t(k_o, &k))
+    return NULL;
+
+  if (igraphmodule_PyObject_to_integer_t(from_o, &from))
```

Comment:
This should be `igraphmodule_PyObject_to_vid` because `from` is a vertex index, but this conversion function also allows the user to use a vertex _name_ and it will properly be resolved to the corresponding index.